core/src/engine.cpp

```diff
@@ -0,0 +1,34 @@
+
+#include "knoting/engine.h"
+
+namespace knot {
+
+Engine::Engine() {
+    m_windowModule = std::make_shared<knot::Window>(m_windowWidth, m_windowHeight, m_windowTitle);
+
+    m_engineModules.emplace_back(m_windowModule);
+
+    for (auto& module : m_engineModules) {
+        module->on_awake();
+    }
+}
+
+void Engine::update_modules() {
+    for (auto& module : m_engineModules) {
+        module->on_update(0);
+    }
+
+    m_windowModule->draw();
```

Comment:
Should we add another call `draw_modules` so we have it as a separate stage, even though most of the time they will be called together?

At the moment only windows needs the draw call, but other modules/systems might need it in the future.

Reply:
when the render module is implemneted it will have a on_update method attached that draws all of the commands submitted to it via various shader piplines